src/queue.cpp

```diff
@@ -63,6 +64,10 @@ cvk_command_queue::cvk_command_queue(
 
     TRACE_CNT(batch_in_flight_counter, 0);
     TRACE_CNT(group_in_flight_counter, 0);
+
+    if (init_image_queue) {
+        m_context.reset(nullptr);
+    }
```

Comment:
Why is this required?

Reply:
`cvk_context` is refcounted. It will not be freed until every object releases their hold on it.
`cvk_command_queue` has a ref on `cvk_context` and that new queue for image initialisation is freed in `cvk_context` destructor with that PR.
Thus to allow everything to get freed properly, the new queue for image initialisation releases the context at creation.

Reply:
That makes sense, thanks. Hmm, rather than introducing a new type of queue effectively, with state in every single command queue, we could introduce a `cvk_command_queue::detach_from_context` function doing the above. That function would only be called in `get_or_create_image_init_command_queue` and you could just assert that `m_context` is non-NULL in `get_or_create_printf_buffer` (although this should really never be called if the command queue is only used for image init; part of me thinks this assert could be omitted, the resulting crash would be very obvious and we wouldn't run the risk of adding new functions that use `m_context` without a corresponding assert). WDYT?

Reply:
I'm fine with that.
Done